ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/Topic.java

```diff
@@ -58,11 +87,39 @@ public SerdeSupplier getSerdeSupplier() {
     return serdeSupplier;
   }
 
+  public SerdeFactory getKeySerdeFactory() {
+    return keySerdeFactory;
+  }
+
   Serializer getSerializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getSerializer(schemaRegistryClient);
   }
 
   Deserializer getDeserializer(final SchemaRegistryClient schemaRegistryClient) {
     return serdeSupplier.getDeserializer(schemaRegistryClient);
   }
+
+  Serializer getKeySerializer() {
+    return keySerdeFactory.create().serializer();
+  }
+
+  @SuppressWarnings("unchecked")
+  Deserializer getKeyDeserializer() {
+    final Serde keySerde = keySerdeFactory.create();
+    if (keySerde instanceof TimeWindowedSerde) {
+      final TimeWindowedSerde windowedSerde = (TimeWindowedSerde) keySerde;
+      final TimeWindowedDeserializer timeWindowedDeserializer =
+          (TimeWindowedDeserializer) windowedSerde.deserializer();
+      if (!windowSize.isPresent()) {
+        if (timeWindowedDeserializer.getWindowSize() != Long.MAX_VALUE) {
+          return new TimeWindowedDeserializer(
+              new StringDeserializer(),
+              timeWindowedDeserializer.getWindowSize());
+        }
+        throw new KsqlException("Window size is not present for time windowed deserializer.");
+      }
+      return new TimeWindowedDeserializer(new StringDeserializer(), windowSize.get());
+    }
+    return keySerdeFactory.create().deserializer();
```

Comment:
Is this different from simply returning `keySerde.deserializer()`?

Reply:
It's the same. I changed it to use `keySerde`.